### ISSUE TITLE
Use window.addEventListener communication instead of QR code

### ISSUE THREAD

**Issue by jdkanani** [WalletConnect/walletconnect#14]
If we use `window.addEventListener`, anyone with React/Vue component can listen `walletconnect:session` and emit `walletconnect:dismissed` event to perform different actions.

Currently, there is no way to dismiss Walletconnect session if the user closes QR window.

**Comment by ptsayli** [comment]
@jdkanani added those in provider https://github.com/WalletConnect/walletconnect-web3-provider/pull/2

**Comment by pedrouid** [comment]
Good point @jdkanani, maybe we can move those listeners from the walletconnect/walletconnect-web3-provider to walletconnect/walletconnect instead

What do you think @ptsayli?

**Comment by pedrouid** [comment]
On second thought, why dont we use a different pattern similar to web3 callback promise events. 

https://web3js.readthedocs.io/en/1.0/callbacks-promises-events.html

.on(‘session_created’)
.on(‘session_dismissed’)


<sub>Sent with <a href="http://githawk.com">GitHawk</a></sub>

**Comment by ptsayli** [comment]
Yeah. I will add those events

**Comment by jdkanani** [comment]
This will add `EventEmitter` as dependency and removes `listenForSessionStatus` and `listenForTransactionStatus`.
